cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamFormatCommand.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.command;
+
+import co.cask.cdap.api.data.format.FormatSpecification;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.cli.ArgumentName;
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.util.AbstractAuthCommand;
+import co.cask.cdap.client.StreamClient;
+import co.cask.cdap.internal.io.SchemaTypeAdapter;
+import co.cask.cdap.proto.StreamProperties;
+import co.cask.common.cli.Arguments;
+import com.google.common.base.Splitter;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Sets the Format Specification of a stream.
+ */
+public class SetStreamFormatCommand extends AbstractAuthCommand {
+
+  private static final Gson GSON = new Gson();
+  private final StreamClient streamClient;
+
+  @Inject
+  public SetStreamFormatCommand(StreamClient streamClient, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.streamClient = streamClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String streamId = arguments.get(ArgumentName.STREAM.toString());
+    StreamProperties currentProperties = streamClient.getConfig(streamId);
+
+    String formatName = arguments.get(ArgumentName.FORMAT.toString());
+    Schema schema = getSchema(arguments);
+    Map<String, String> settings = Collections.emptyMap();
+    if (arguments.hasArgument(ArgumentName.SETTINGS.toString())) {
+      settings = Splitter.on(" ").withKeyValueSeparator("=").split(arguments.get(ArgumentName.SETTINGS.toString()));
```

Comment:
Put in some common code which parses key-value input, possibly in `AbstractCommand`. Would be nice to use the same code to parse all key-value input in the CLI, but this can probably be done in another PR.